src/libstd/fs.rs

```diff
@@ -1814,6 +1814,18 @@ mod tests {
         check!(fs::set_permissions(&out, attr.permissions()));
     }
 
+    #[cfg(windows)]
+    #[test]
+    fn copy_file_preserves_streams() {
```

Comment:
To elaborate further on what's going on, I'm creating the file `in.txt` but instead of writing to the default data stream, I'm instead writing to the `bunny` stream.
If I copy the file to a new location it brings all the streams with it, and the number of bytes copied (what I assume the return value from `fs::copy` means) is the same as the number of bytes I wrote to the `bunny` stream since its the only stream with data.
When getting the size of the file, I'm getting that information on the default data stream, which is empty so its size is 0. If I wanted information on the other streams I'd have to explicitly specify them by name.